bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```
@@ -440,24 +444,50 @@ protected Composite createBrowseComposite(Composite parent) {
 		Composite panel = new Composite(parent, SWT.NONE);
 
 		BorderLayout layout = new BorderLayout();
-		layout.marginHeight = convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_MARGIN);
-		layout.marginWidth = convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_MARGIN);
-		layout.spacing = convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_SPACING);
+		layout.marginHeight = convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_MARGIN) / 2;
+		layout.marginWidth = convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_MARGIN) ;
+		layout.spacing = convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_SPACING) / 2;
 		panel.setLayout(layout);
 		panel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
 		panel.setFont(parent.getFont());
 		return panel;
 	}
 
+	private void populateResolvedPathLabel(Label label, String characters) {
+		try {
+			File location = new File(characters);
+			String normalisedPath = ""; //$NON-NLS-1$
+			if (location.exists() && location.isDirectory()) {
+				normalisedPath = location.getAbsoluteFile().toPath().toRealPath(LinkOption.NOFOLLOW_LINKS).normalize()
+						.toString();
+			} else {
+				normalisedPath = location.getAbsoluteFile().toPath().normalize().toString();
+			}
+			String normalisedPathWithSeperator = normalisedPath + File.separator;
+			if (characters.startsWith("~")) //$NON-NLS-1$
+			{
+				label.setText(IDEWorkbenchMessages.ChooseWorkspaceDialog_TildeNonExpandedWarning0
+						+ IDEWorkbenchMessages.ChooseWorkspaceDialog_ResolvedAbsolutePath0 + System.lineSeparator()
+						+ location.getAbsoluteFile().toPath().normalize());
+			} else if (!characters.equalsIgnoreCase(normalisedPath)
+					&& !characters.equalsIgnoreCase(normalisedPathWithSeperator)) {
+				label.setText(IDEWorkbenchMessages.ChooseWorkspaceDialog_ResolvedAbsolutePath0 + System.lineSeparator()
+						+ location.getAbsoluteFile().toPath().normalize());
+			} else {
+				label.setText(""); //$NON-NLS-1$
+			}
+		} catch (Exception ex) {
```

ignoring exception forgets to update label